src/contracts/core/DelegationManager.sol

```diff
@@ -264,8 +278,72 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
+     * This is because undelegation limbo is subject to the same restrictions as completing a withdrawal, to ensure that a staker cannot use undelegation
+     * limbo to avoid potentially being subject to slashing.
+     * @dev Note that this checks against the frozen status of the operator *that the caller was delegated to when they entered undelegation limbo*, to provide
+     * the same kind of slashing eligibility as withdrawals via the StrategyManager or DelegationManager.
+     */
+    function exitUndelegationLimbo(
+        uint256 middlewareTimesIndex
+    ) external onlyNotFrozen(_stakerUndelegationLimboStatus[msg.sender].delegatedAddress) {
+        require(
+            isInUndelegationLimbo(msg.sender),
+            "DelegationManager.exitUndelegationLimbo: must be in undelegation limbo"
+        );
 
-        return withdrawalRoot;
+        uint32 limboStartBlock = _stakerUndelegationLimboStatus[msg.sender].startBlock;
+        require(
+            slasher.canWithdraw(
+                _stakerUndelegationLimboStatus[msg.sender].delegatedAddress,
+                limboStartBlock,
+                middlewareTimesIndex
+            ),
+            "DelegationManager.exitUndelegationLimbo: shares in limbo are still slashable"
+        );
+
+        // enforce minimum delay lag
+        require(
+            limboStartBlock + strategyManager.withdrawalDelayBlocks() <= block.number,
+            "DelegationManager.exitUndelegationLimbo: withdrawalDelayBlocks period has not yet passed"
+        );
+
+        // delete the pod owner's undelegation limbo details and emit an event
+        delete _stakerUndelegationLimboStatus[msg.sender];
+        emit UndelegationLimboExited(msg.sender);
+
+        // return any shares that the staker has to the delegation system
+        address operator = delegatedTo[msg.sender];
+        uint256 podShares = eigenPodManager.podOwnerShares(msg.sender);
+        if (podShares != 0) {
+            _increaseOperatorShares({
+                operator: operator,
+                staker: msg.sender,
+                strategy: beaconChainETHStrategy,
+                shares: podShares
+            });
+        }
+        bool stakerHasSharesInStrategyManager = (strategyManager.stakerStrategyListLength(msg.sender) != 0);
+        if (stakerHasSharesInStrategyManager) {
+            (IStrategy[] memory strategies, uint256[] memory strategyShares) = strategyManager.getDeposits(msg.sender);
+
+            for (uint256 i = 0; i < strategies.length; ) {
+                _increaseOperatorShares({
+                    operator: operator,
+                    staker: msg.sender,
+                    strategy: strategies[i],
+                    shares: strategyShares[i]
+                });
+
+                unchecked {
+                    ++i;
+                }
+            }
+        }
```

Comment:
I've realized that this logic is actually entirely unnecessary -- since we are enforcing that a staker cannot re-delegate while in undelegation limbo, `operator` here should always be the zero address, meaning no shares should ever get delegated here!

Reply:
removed the logic in this commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/208/commits/f6164740c93e6b8360d536c17fd91f05936e5f62